pkg/monitortests/testframework/uploadtolokiserializer/monitortest.go

```diff
@@ -25,13 +26,15 @@ import (
 )
 
 type lokiSerializer struct {
+	monitorStartTime time.Time
 }
 
 func NewUploadSerializer() monitortestframework.MonitorTest {
 	return &lokiSerializer{}
 }
 
 func (w *lokiSerializer) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	w.monitorStartTime = time.Now()
```

Comment:
Immediately before the call to [`monitorTestRegistry.StartCollection()`](https://github.com/openshift/origin/blob/11837c1460a13e900603a1dbafaea6525276aabf/pkg/monitor/monitor.go#L65), we get the current time and use that as the `Monitor.startTime`.  `MonitorTestRegistry.StartCollection` calls `startCollectionWithPanicProtection` for each `monitorTest` which calls `monitorTest.StartCollection which is when line 37 above is called.  So, `time.Now()` is when the data collection starts and about he same time as the monitor's `startTime` which makes it so I don't have to pass it.

Reply:
Drop this in as a comment plz, this is good stuff.